### PR TITLE
Login iframe using token from hydra

### LINKED ISSUE #6: I can Login into the admin ui and see a preview of a private pages (pass auth token)
## Description

As an editor, I want to Login into the admin UI and see the toolbar appear so that I can be authenticated to view/edit content.

## Technical Approach

Since our first approach is building an addon for volto, so setting up an addon using `cookieplone` which will be a good starting point , we can depend on volto to provide UI for logging in and let Plone authenticate the user (editor). 
- The edit URL and view URL are two different things

Follow up
- [ ] #47

### PR DESCRIPTION
**Pull request by MAX-786**
Fixes #6 
Needs #38 to be merged first.

### Approach

In Volto-Hydra, Iframe component retrieves `auth_token` from redux store and  listens to message event from frontend and sends back the token to it. When the user LOGIN/LOGOUT or token is RENEWED, volto updates the store with the updated token.

In hydra.js, it initiates the Bridge, and starts listening to the token response from the Hydra. It also have the method `(getTokenFromCookies)` to fetch the token from the cookies and pass it to the integrator to use it in the `ploneClient.initialize()`.

### How to Integrate your frontend 

- Add 'hydra.js` in your frontend.
- Initialize the Bridge using `initBridge` method provided by './hydra.js', use 'https://hydra.pretagov.com` for option `adminOrigin` to tryout demo.
- Use the getToken() method provided by './hydra.js' to get the auth_token from Hydra. Use this in your plone client.
- At [Volto-Hydra demo](https://hydra.pretagov.com/) type in your hosted frontend url  to preview public content and login to see the private pages.

Example Usage:
```js
// nextjs 14
import ploneClient from "@plone/client";
import { useQuery } from "@tanstack/react-query";
import { initBridge } from "@/utils/hydra";

export default function Blog({ params }) {
  const bridge = initBridge("http://localhost:3000"); // Origin of your local Volto-Hydra
  const token = bridge._getTokenFromCookie();
  const client = ploneClient.initialize({
    apiPath: "http://localhost:8080/Plone/", // Plone backend
    token: token,
  });
  const { getContentQuery } = client;
  const { data, isLoading } = useQuery(getContentQuery({ path: '/blogs' }));

  if (isLoading) {
    return <div>Loading...</div>;
  }
  return (
    <div> {data.title}</div>
  )


```

### CODE REVIEW

**Comment by djay** [comment]
@MAX-786 the better way to deal with a PR that is dependent on another PR is to base the PR on that other PR. then the diffs show whats actually changed and it's obvious the relationship.
It doesn't mean you have to merge this PR into the other one fist. Once the base PR is merged you just change the base of this PR again. it may even do that automatically if you delete that branch of the base PR.

image

**Comment by djay** [comment]
@MAX-786 this PR changes what an integrator is supposed to do with the API. So include the changes to the README with this PR. Keep them together. it's much easier to understand.

**Comment by djay** [comment]
More to that point, it's just as easy to read in a diff to the readme as it is in the comment on the PR. but the difference is you have to repeat it again anyway in the readme later so just put it there first. 
I am a believer in documentation driven development also. What this means is you try to write the documetnation first before you impliment the code. In this case it would mean the start of the PR is commits to change the README to show what you are about to implement. then implement. then make any tweaks to the readme at hte end. 
Why is this a good idea? Because it helps you focus on the user and your goal. In this case, even though the stories are written from editor perspective, your goal is to implement each editor function with the least lines of added integrator documenations as possible. The simplest and clearest set of instructions to add the required editor functionality. The more things that are automatic or optional the better. Optional means that you only have to do this more complicated API call to make the experience a bit better but the story is covered by doing nothing or something simpler.

File: README.md [hunk 1]
```diff
@@ -94,22 +94,61 @@ TODO: more integrations will be added below as the [Hydra GSoC project progresse
 
 #### Authenticate frontend to access private content
 
-After initializing the Bridge, you can import `get_token` method from [hydra.js](https://github.com/collective/volto-hydra/tree/hydra.js). This will return the token which you can use to access private content from Plone backend.
```
Comment (djay) [inline README.md:97]: @MAX-786 this assumes you have to have the bridge loaded all the time even when you aren't using the editor. One goal of a decoupled frontend is to make it as fast and as light as possible. So we need to provide a way for them to know when the editor is being used without having to have the bridge js loaded all the time. This is why I suggest its better to pass the token via the url argument and that signifies the editor is now being used and the frontend should load and init the bridge.

File: README.md [hunk 2]
```diff
@@ -94,22 +94,61 @@ TODO: more integrations will be added below as the [Hydra GSoC project progresse
...
+
+You can go to content view, to delete single or multiple pages at once.
+
+#### Edit a page
+
+You can click on edit while previewing that page to enter see sidebar appear. You can change title of the page.
+
+TODO: editing of Blocks/contents of a page is not implemented yet.
+
+Reference Issue: [#13](https://github.com/collective/volto-hydra/issues/13)
```
Comment (djay) [inline README.md:151]: I don't think its important to list out this much details. You can just say

---
Now your editors can navigate the site within the editor or in the frontend displayed in the middle of the screen, add, remove and do normal plone toolbar functions as well as edit a page metadata via the sidebar.

**Review by djay — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (MAX-786) [reply README.md:97]: Yep, I was looking into your previous comments about this. I'll fix it soon.

**Comment by djay** [comment]
already got merged in via c4f12ecfde462bb2660679c7bc197b04da57ed36

**Pull request closed without merging**